gstack-buffer/src/lib.rs

```diff
@@ -0,0 +1,94 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Stack allocations utils.
+
+#![feature(c_unwind)]
+#![no_std]
+
+extern crate alloc;
+
+use alloc::vec::Vec;
+
+use core::{
+    ffi::c_void,
+    mem::{ManuallyDrop, MaybeUninit},
+    slice,
+};
+
+const MAX_BUFFER_SIZE: usize = 64 * 1024;
+
+type Callback = unsafe extern "C-unwind" fn(ptr: *mut MaybeUninit<u8>, data: *mut c_void);
+
+#[cfg(any(
+    feature = "compile-alloca",
+    all(not(feature = "compile-alloca"), target_arch = "wasm32")
+))]
+extern "C-unwind" {
+    fn c_with_alloca(size: usize, callback: Callback, data: *mut c_void);
+}
+
+#[cfg(all(not(feature = "compile-alloca"), not(target_arch = "wasm32")))]
+unsafe extern "C-unwind" fn c_with_alloca(_size: usize, callback: Callback, data: *mut c_void) {
```

Comment:
put unsafes in-place pls

Reply:
This function is kind of a polyfill when we can't use the pre-built library. Since the c_with_alloca function from C is unsafe, this polyfill should also be marked as unsafe.